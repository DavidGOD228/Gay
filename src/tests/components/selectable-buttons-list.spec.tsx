import 'react-native'
// Note: test renderer must be required after react-native.
import React from 'react'
import renderer from 'react-test-renderer'
import SelectableButtonList from '../../components/selectable-buttons-list'

it('renders correctly when has options', () => {
  const optionsArray = [
    {
      value: 'Guy Nex Dor',
      isSelected: false,
      key: 1
    },
    {
      value: 'All-American',
      isSelected: false,
      key: 2
    },
    {
      value: 'Suit & Tie',
      isSelected: false,
      key: 3
    },
    {
      value: 'Bear',
      isSelected: false,
      key: 4
    },
    {
      value: 'Muscle',
      isSelected: false,
      key: 5
    },
    {
      value: 'Geek/Nerd',
      isSelected: false,
      key: 6
    }
  ]
  const tree = renderer.create(
    <SelectableButtonList items={optionsArray}
                          onItemSelected={item => item.isSelected = !item.isSelected}/>
  )
  expect(tree)
    .toMatchSnapshot()
})

it('renders correctly when has no options', () => {
  const optionsArray = []
  const tree = renderer.create(
    <SelectableButtonList items={optionsArray}
                          onItemSelected={item => item.isSelected = !item.isSelected}/>
  )
  expect(tree)
    .toMatchSnapshot()
})
