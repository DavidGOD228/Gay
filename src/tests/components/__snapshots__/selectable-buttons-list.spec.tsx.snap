// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when has no options 1`] = `null`;

exports[`renders correctly when has options 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#A5B4BD",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      Guy Nex Dor
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#A5B4BD",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      All-American
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#A5B4BD",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      Suit & Tie
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#A5B4BD",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      Bear
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#A5B4BD",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      Muscle
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#A5B4BD",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      Geek/Nerd
    </Text>
  </View>
</View>
`;
